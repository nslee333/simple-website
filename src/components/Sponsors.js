import ladd_group from "../../public/images/ladd_group_logo.jpg";
import tao from "../../public/images/tao_logo.png";
import uptech from "../../public/images/uptech_logo.png";
import codepen from "../../public/images/codepen_logo.png";
import Image from "next/image";

export default function Sponsors() {
  return (
    <>
      <div className="bg-white p-10">
        <div>
          <h5 className="mb-5 flex justify-center text-black xs:text-2xl md:text-3xl">
            Thank you so much to our incredible sponsors!
          </h5>
        </div>
        <div className="">
          <div
            className="flex 
                        justify-around 
                        xs:flex-wrap 
                        xs:pb-5 
                        md:pb-14 
                        md:pt-20"
          >
            <a href="https://www.uptechstudio.com/" target="_blank">
              <Image
                src={uptech}
                className="m-5 
                          aspect-auto 
                          xs:w-[200px] 
                          md:w-[260px]"
                alt="uptech logo"
              />
            </a>
            <a href="https://www.techoregon.org/" target="_blank">
              <Image
                src={tao}
                className="m-5 aspect-auto xs:w-[200px] md:w-[260px]"
                alt="tech oregon logo"
              />
            </a>
            <a href="https://codepen.io/" target="_blank">
              <Image
                src={codepen}
                className="mb-5 ml-5 mr-5 mt-8 aspect-auto xs:w-[200px] md:w-[260px] "
                alt="codepen logo"
              />
            </a>
            <a href="https://www.bendpropertysource.com/" target="_blank">
              <Image
                src={ladd_group}
                className="mb-5 
                            ml-5 
                            mr-5 
                            aspect-auto 
                            xs:mb-0 
                            xs:mt-5 
                            xs:w-[200px] 
                            md:mt-2 
                            md:w-[260px]"
                alt="ladd group logo"
              />
            </a>
          </div>
          <div className="flex justify-center">
            <div>
              <h5 className="mt-5 text-2xl italic md:text-3xl">
                We are always looking for new sponsors
              </h5>
              <div className="flex justify-center">
                <a
                  htmlFor="button"
                  className="shadow-grey 
                              mt-5 
                              rounded 
                              bg-[#7FE7D7] 
                              p-3 
                              text-xl 
                              shadow 
                              hover:origin-bottom 
                              hover:translate-y-0.5 
                              hover:shadow-none 
                              md:text-2xl"
                  href="#contact"
                >
                  Contact Us
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
}
