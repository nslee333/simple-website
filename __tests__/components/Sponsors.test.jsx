import { render, screen } from "@testing-library/react";
import { toHaveAttribute } from "@testing-library/jest-dom";
import Sponsors from "../../src/components/Sponsors";
import renderer from "react-test-renderer";

test("Sponsors renders without crashing", () => {
  render(<Sponsors />);
});

test("Sponsors renders correctly according to snapshot.", () => {
  const tree = renderer.create(<Sponsors />).toJSON();

  expect(tree).toMatchSnapshot();
});

test("Uptech logo has link to uptechstudio.com", () => {
  render(<Sponsors />);

  expect(screen.getByRole("link", { name: "uptech logo" })).toHaveAttribute(
    "href",
    "https://www.uptechstudio.com/"
  );
});

test("tech oregon logo has link to techoregon.org", () => {
  render(<Sponsors />);

  expect(
    screen.getByRole("link", { name: "tech oregon logo" })
  ).toHaveAttribute("href", "https://www.techoregon.org/");
});

test("Codepen logo has link to codepen.io", () => {
  render(<Sponsors />);

  expect(screen.getByRole("link", { name: "codepen logo" })).toHaveAttribute(
    "href",
    "https://codepen.io/"
  );
});

test("Ladd group logo has link to bendpropertysource.com", () => {
  render(<Sponsors />);

  expect(screen.getByRole("link", { name: "ladd group logo" })).toHaveAttribute(
    "href",
    "https://www.bendpropertysource.com/"
  );
});

test("Contact Us has link to contact form", () => {
  render(<Sponsors />);

  expect(screen.getByRole("link", { name: "Contact Us" })).toHaveAttribute(
    "href",
    "#contact"
  );
});
