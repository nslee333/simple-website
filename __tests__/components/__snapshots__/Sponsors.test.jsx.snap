// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sponsors renders correctly according to snapshot. 1`] = `
<div
  className="bg-white p-10"
>
  <div>
    <h5
      className="mb-5 flex justify-center text-black xs:text-2xl md:text-3xl"
    >
      Thank you so much to our incredible sponsors!
    </h5>
  </div>
  <div
    className=""
  >
    <div
      className="flex  justify-around  xs:flex-wrap  xs:pb-5  md:pb-14  md:pt-20"
    >
      <a
        href="https://www.uptechstudio.com/"
        target="_blank"
      >
        <img
          alt="uptech logo"
          className="m-5  aspect-auto  xs:w-[200px]  md:w-[260px]"
          data-nimg="1"
          decoding="async"
          height={40}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={40}
        />
      </a>
      <a
        href="https://www.techoregon.org/"
        target="_blank"
      >
        <img
          alt="tech oregon logo"
          className="m-5 aspect-auto xs:w-[200px] md:w-[260px]"
          data-nimg="1"
          decoding="async"
          height={40}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={40}
        />
      </a>
      <a
        href="https://codepen.io/"
        target="_blank"
      >
        <img
          alt="codepen logo"
          className="mb-5 ml-5 mr-5 mt-8 aspect-auto xs:w-[200px] md:w-[260px] "
          data-nimg="1"
          decoding="async"
          height={40}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={40}
        />
      </a>
      <a
        href="https://www.bendpropertysource.com/"
        target="_blank"
      >
        <img
          alt="ladd group logo"
          className="mb-5  ml-5  mr-5  aspect-auto  xs:mb-0  xs:mt-5  xs:w-[200px]  md:mt-2  md:w-[260px]"
          data-nimg="1"
          decoding="async"
          height={40}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={40}
        />
      </a>
    </div>
    <div
      className="flex justify-center"
    >
      <div>
        <h5
          className="mt-5 text-2xl italic md:text-3xl"
        >
          We are always looking for new sponsors
        </h5>
        <div
          className="flex justify-center"
        >
          <a
            className="shadow-grey  mt-5  rounded  bg-[#7FE7D7]  p-3  text-xl  shadow  hover:origin-bottom  hover:translate-y-0.5  hover:shadow-none  md:text-2xl"
            href="#contact"
            htmlFor="button"
          >
            Contact Us
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
